iothub/device/src/MethodInvokeRequest.cs

```diff
@@ -23,16 +23,19 @@ internal class MethodInvokeRequest
         /// <exception cref="ArgumentException">If <b>methodName</b> is null or whitespace</exception>
         public MethodInvokeRequest(string methodName, string payload, TimeSpan? responseTimeout, TimeSpan? connectionTimeout)
         {
-            ValidatePayloadIsJson(payload);
+            if (!string.IsNullOrEmpty(payload))
+            {
+                ValidatePayloadIsJson(payload);
+                this.Payload = new JRaw(payload);
```

Comment:
Is Leaving this.Payload == null ok?

Reply:
Yes there are checks for Payload being null in other parts of the code.